src/python/deepgnn/graph_engine/snark/tests/convert_test.py

```diff
@@ -720,6 +775,255 @@ def test_sanity_edge_sparse_features_data(graph_with_sparse_features):
+        npt.assert_equal(np.frombuffer(result[32:48], dtype=np.uint64), [3, 4])
+        npt.assert_almost_equal(
+            np.frombuffer(result[48:56], dtype=np.float32), [3.3, 4.4]
+        )
+        npt.assert_equal(np.frombuffer(result[56:64], dtype=np.int32), [3, 4])
+        npt.assert_equal(np.frombuffer(result[64:80], dtype=np.uint64), [5, 6])
+        npt.assert_almost_equal(
+            np.frombuffer(result[80:88], dtype=np.float32), [5.5, 6.6]
+        )
+        npt.assert_equal(np.frombuffer(result[88:96], dtype=np.int32), [5, 6])
+
+
+def test_edge_list_header_multiple_partitions():
+    output = tempfile.TemporaryDirectory()
+    data_data = [
+        "0,-1\n",
+        "1,-1\n",
+        "2,-1\n",
+    ]
+    meta_data = {"default_node_type": 0, "default_node_weight": 1.5}
+    _gen_edge_list(output, data_data, meta_data, partitions=2)
+    with open("{}/node_{}_{}.map".format(output.name, 0, 0), "rb") as nm:
+        expected_size = 2 * (2 * 8 + 4)
+        result = nm.read(expected_size + 8)
+        assert len(result) == expected_size
+        assert result[0:8] == (0).to_bytes(8, byteorder=sys.byteorder)
+        assert result[8:16] == (0).to_bytes(8, byteorder=sys.byteorder)
+        assert result[16:20] == (0).to_bytes(4, byteorder=sys.byteorder)
+        assert result[20:28] == (2).to_bytes(8, byteorder=sys.byteorder)
+        assert result[28:36] == (1).to_bytes(8, byteorder=sys.byteorder)
+        assert result[36:40] == (0).to_bytes(4, byteorder=sys.byteorder)
+
+
+def test_edge_list_error_checking():
+    decoder = EdgeListDecoder()
+    with pytest.raises(StopIteration):
+        next(decoder.decode(""))
+    with pytest.raises(ValueError):
+        next(decoder.decode("x"))
+    with pytest.raises(RuntimeError):
+        next(decoder.decode("0,-1"))
+    with pytest.raises(RuntimeError):
+        next(decoder.decode("0,-1,4"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,x"))
+    # ignores feature vector if fails to read
+    next(decoder.decode("0,-1,4,1,bad_key"))
+    next(decoder.decode("0,-1,4,1,float32"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2,1"))
+    with pytest.raises(ValueError):
+        next(decoder.decode("0,-1,4,1,float32,2,1,x"))
+    next(decoder.decode("0,-1,4,1,float32,2,1,1"))
+    next(decoder.decode("0,-1,4,1,binary,1,test"))
+    next(decoder.decode("0,-1,4,1,float32,0"))
+    with pytest.raises(RuntimeError):
+        gen = decoder.decode("0,1")
+        next(gen)
+    with pytest.raises(RuntimeError):
+        gen = decoder.decode("0,1,4")
+        next(gen)
+    gen = decoder.decode("0,1,4,1")
+    next(gen)
+
+
+def test_edge_list_binary_escape():
+    decoder = EdgeListDecoder()
+    src, dst, typ, weight, features = next(
+        decoder.decode("0,-1,0,1.0,binary,1,test\,feature")
```

Comment:
Can you add a few more test cases:
* "0,-1,0,1.0,binary,1,test\,feature"
* "0,-1,0,1.0,binary,1,test\\,feature"
* "0,-1,0,1.0,binary,1,test\\\,feature"
* "0,-1,0,1.0,binary,1,test\,feature\"
* "0,-1,0,1.0,binary,1,test\,feature\,"
* "0,-1,0,1.0,binary,1,test\,feature,"
* "0,-1,0,1.0,binary,1,test\,\,feature"
* "0,-1,0,1.0,binary,1,\test,\\feature"
* "0,-1,0,1.0,binary,1,\,feature"

Reply:
Sorry, there are more test cases, md has it's own escape rules..